File: content/03/code/dotnet/06--step06-sketch-out-parameter-type/Shouty/Shouty.Specs/Support/ParameterTypes.cs
using System;
using TechTalk.SpecFlow;

namespace Shouty.Specs.Support
{
    [Binding]
    public class ParameterTypes
    {
        [StepArgumentTransformation]
        public Person ConvertPerson(string name)
        {
            return new Person(name);
        }
    }
}

File: content/bdd-with-cucumber/09/code/java/04-introduce-person-paramter-type/shouty/src/test/java/shouty/support/ParameterTypes.java
package shouty.support;

import io.cucumber.java.ParameterType;
import shouty.Person;

public class ParameterTypes {
    private final ShoutyWorld world;

    public ParameterTypes(ShoutyWorld world) {
        this.world = world;
    }

    @ParameterType("Lucy|Sean|Larry")
    public Person person(String name) {
        return new Person(name, world.network, 0);
    }
}
